lightly/embedding/embedding.py

```diff
@@ -107,20 +107,23 @@ def embed(self,
         embeddings, labels, fnames = None, None, []
 
         if lightly._is_prefetch_generator_available():
-            pbar = tqdm(
-                BackgroundGenerator(dataloader, max_prefetch=3),
-                total=len(dataloader)
-            )
-        else:
-            pbar = tqdm(dataloader, total=len(dataloader))
+            dataloader = BackgroundGenerator(dataloader, max_prefetch=3),
+        
+        pbar = tqdm(
+            dataloader,
+            total=len(dataloader.dataset),
+            unit='imgs'
+        )
 
         efficiency = 0.0
         embeddings = []
         labels = []
         with torch.no_grad():
 
             start_timepoint = time.time()
-            for (img, label, fname) in pbar:
+            for (img, label, fname) in dataloader:
+
+                bsz = img.shape[0]
```

Comment:
why are you pulling out the with of an image? 🤔

Reply:
`img` is a batch of images so `img.shape[0]` gives the batch size. But this line is actually not necessary as `batch_size = img.shape[0]` is already defined a bit further down (line 132)

Reply:
Because img is actually a list of images (one batch), but badly named.

Reply:
Furthermore, this line is duplicate to `batch_size = img.shape[0]` a few lines below -> delete it.